tls/s2n_client_extensions.c

```diff
@@ -26,9 +26,9 @@
 
 #include "utils/s2n_safety.h"
 #include "utils/s2n_blob.h"
+#include "tls/s2n_client_extensions.h"
```

Comment:
I don't see anything external calling your refactored functions in this change. Is that in a later change? Why do we need to add it to a header?